resources/assets/js/pages/home.js

```diff
@@ -1,14 +1,74 @@
 (function() {
     var showDialogButton = document.querySelector('#fire'),
-        formDeploy = document.querySelector('form');
+        formDeploy = document.querySelector('form'),
+        serverSelect = document.querySelector('#select-server'),
+        branchSelect = document.querySelector('#select-branch'),
+        taskSelect = document.querySelector('#select-task'),
+        progressBar = document.querySelector('#progress-bar');
 
     if (showDialogButton) {
         var dialog = new DeployerConfirmDialog();
 
-        showDialogButton.addEventListener('click', function() {
+        showDialogButton.addEventListener('click', function(event) {
+            event.preventDefault();
+
+            if(serverSelect.value === "-1") {
```

Comment:
I use -1 because always it will select the default element ("Select a server" and so on). So even if I don't set the value on the `option` tag, it will fill the value with the innerHtml of the option.

:+1: for select